### ISSUE TITLE
Stake Drop Workflow in Web Wallet

### ISSUE THREAD

**Issue by think-in-universe** [near/near-wallet-roadmap#23]
**Summary**

To help a user understand how NEAR work as a Proof of Stake blockchain, it’s important to guide the user through the staking process via tooltip, to have the user prace how to staking via our wallet UI, once upon a new account is created.

**Intended Outcome**

1. Through the guided staking journey, the new users (holders) learnt why staking is important and beneficial to themselves, and how to complete staking via web wallet UI; (Metric: the ratio of users who know how to stake)
2. The on-boarding experience is actually better since the user learnt more about how the wallet works once joined. This education process may not only include staking procedure but also other tabs such as wallet and account if necessary. (Metric: the user gets familiar with the features of the wallet in a few minutes. Now we often receive a lot of questions about wallet from our users...)
3. The total ratio of staking is increased as a result as more users are incentivized to stake. 

**How will it work?**

1. A url parameter such as `guide`, will be used to indicate that the account creation process requires a guided process after the account creation is completed, e.g. `https://wallet.near.org/create/<key>?guide=staking` . there could be multiple categories as the parameter of `guide` such as staking, transfer, account, etc. 
2. Tools like https://www.trychameleon.com/ (or any other applicable tools) could be used to integrate the education process into the wallet UI. (this might only be relevant if we cant have a team to build the guide natively)
3. Users could receive some extra rewards and incentives, etc.after they complete the staking process
4. More details about the user workflow can be found in the next section. 

**When you would like to see it released?**

In April or May, we’d like to do some aggressive airdrop campaign, which means ideally there should be at least a smaller one in March to collect a bit of data to iterate further. So the recommended roadmap would be:

1. In mid of April, a minimal version that could be used for collecting feedback and testing; 
2. By end of April, we have a relatively mature one that could be deployed into production.

Acceptance Criteria

1. The guided stake drop process is included into the web wallet UI;
2. The user should typically spend less than 2 minutes understanding why staking is helpful and how to do staking in wallet UI;
3. The performance of the web wallet (e.g. page loading time) doesn’t decrease apparently due to introducing the new feature. 
4. The user experience of the guided staking process is intuitive and smooth.

**Comment by jimmy3dita** [comment]
Robert,

Thank you for this valuable submission!

Conceptually, I agree with the importance of this feature, as the current staking component is not meant to "guide" the user in their choice - it's designed just for mechanically staking tokens to a validator that was already chosen via other channels.

## General feedback
Operationally, I face two main issues:
1. review and validation of the UX/UI
2. quality assurance and code testing

** 1. review UX/UI**
We need help from @corwinharrell, and the timing is too tight to complete the first no-code release of the feature.
It would also help to know the estimated number of impacted users (feel free to keep it confidential too!), and the expected amount staked by each of them.
We may find comparable value for other use cases (e.g. fungible tokens or the Bridge) and follow that direction instead

**2. quality assurance and code testing**
NEAR Wallet team is committed to improving our "production readiness", limiting the number of new features until we have solid regression testing and the existing features (e.g. fungible tokens) cover users' needs and expectations.àà

## Suggested alternative
NEAR Wallet supports the login via third-party apps, and can also sign transactions formed outside of its UI. See how [BerryClub](https://berryclub.io/) or [Paras](https://paras.id/) work for good examples.

The idea here is: how might we build an external app, similar to the Red Packet application, that can guide new users as proposed above, involving the wallet UI only to sign the final staking transaction?

**Comment by think-in-universe** [comment]
Thanks for the feedback Stefano,

Regarding the impact of the feature, in order to help with coming multiple airdrop campaigns, most airdrop (linkdrop) links (let's say 80% or more, Amos could provide more details) would include the `?guide=staking` parameter, so the wallet will guide the user through the staking UI after he/she joined. 

Yan (@picturepan2, the author of [redpacket](https://redpacket.near.org/) ) is working on one prototype for the tooltips with web wallet. Yan is familiar with the wallet code and should be able to deliver with high quality. l agree we don't want to make the wallet app too fat and only keep the core and necessary features. 

For the idea of a standalone "**Staking**" app (or included it into redpacket if our wallet allows redirect after creating a new account), I agree we can have a try. The core requirement here is to **educate holders about why and how to stake, and incentivize them to stake**. To guide the user through the web wallet UI is the most straightforward approach in my opinion, otherwise the new users have to learn about the wallet UI again after they login.

**Comment by think-in-universe** [comment]
We have created one bounty related to this feature: https://github.com/near/bounties/issues/51

**Comment by think-in-universe** [comment]
btw, we already have some existing staking DApps such as https://staking.dokia.cloud/staking/near/validators

**Comment by jimmy3dita** [comment]
> btw, we have some standalone staking DApps such as https://staking.dokia.cloud/staking/near/validators

I'm not sure they support lockup accounts, but it's a great example.

From my point of view, the missing piece is not around the functionality and "where to click", it's rather which validator from the list of 60 (in the future 100, if not more). 
This is not a UI problem that can be solved with tooltips and tutorials, and I believe that most of the work might be done by a well-written article that introduces all differentiation factors of the validators (location, hardware, SRE team, track record) and how to pick the right one.
At that point, we might use a banner linking the user to a "Staking Onboarding tutorial", similar to the one that links to `awesomenear` from the dashboard.

Then, if we see a high drop rate from the staking process, I'm happy to add UX and UI elements that better guide the user in the process.

What do you think? cc @corwinharrell

**Comment by think-in-universe** [comment]
@jimmy3dita yeah. the problem "who" to stake is another thing we need to figure out. 

I think here we mainly want to address the problem of "why" and "how" especially for the users who knew little about staking in NEAR. Before the users learn more about the validators in depth, they could get their hands dirty by trying how staking works in NEAR, by following the tooltips, and staking with a small amount of NEAR.

**Comment by think-in-universe** [comment]
@stefanopepe @corwinharrell for the timeline of the feature, is it possible we move it faster to this month?

We already have a bounty for the feature https://github.com/near/bounties/issues/51 and Yan has been working on it. 

> Yan (@picturepan2, the author of [redpacket](https://redpacket.near.org/) ) is working on one prototype for the tooltips with web wallet. Yan is familiar with the wallet code and should be able to deliver with high quality.

**Comment by stefanopepe** [comment]
@think-in-universe the number of changes in the current UI is too high to accomplish anything within this quarter.

If you need this one done within May, it's better to have it designed and built as a separated staking app, using the same authorization pattern on Dokia's staking app https://staking.dokia.cloud/
